python_modules/libraries/dagster-snowflake/dagster_snowflake/resources.py

```diff
@@ -335,11 +344,249 @@ def write_parquet_file(context):
+            "URI for the OCSP response cache file.  By default, the OCSP response cache "
+            "file is created in the cache directory."
+        ),
+    )
+
+    validate_default_parameters: bool = Field(
+        default=False,
+        description=(
+            "False by default. Raise an exception if either one of specified database, "
+            "schema or warehouse doesn't exists if True."
+        ),
+    )
+
+    paramstyle: Optional[str] = Field(
+        default=None,
+        description=(
+            "pyformat by default for client side binding. Specify qmark or numeric to "
+            "change bind variable formats for server side binding."
+        ),
+    )
+
+    timezone: Optional[str] = Field(
+        default=None,
+        description=(
+            "None by default, which honors the Snowflake parameter TIMEZONE. Set to a "
+            "valid time zone (e.g. America/Los_Angeles) to set the session time zone."
+        ),
+    )
+
+    connector: Optional[str] = Field(
+        default=None,
+        description=(
+            "Indicate alternative database connection engine. Permissible option is "
+            "'sqlalchemy' otherwise defaults to use the Snowflake Connector for Python."
+        ),
+        is_required=False,
+    )
+
+    cache_column_metadata: Optional[str] = Field(
+        default=None,
+        description=(
+            "Optional parameter when connector is set to sqlalchemy. Snowflake SQLAlchemy takes a"
+            " flag cache_column_metadata=True such that all of column metadata for all tables are"
+            ' "cached"'
+        ),
+    )
+
+    numpy: Optional[str] = Field(
+        default=None,
+        description=(
+            "Optional parameter when connector is set to sqlalchemy. To enable fetching "
+            "NumPy data types, add numpy=True to the connection parameters."
+        ),
+    )
+
+    authenticator: Optional[str] = Field(
+        default=None,
+        description="Optional parameter to specify the authentication mechanism to use.",
+    )
+
+    @validator("paramstyle")
+    def validate_paramstyle(cls, v):
+        valid_config = ["pyformat", "qmark", "numeric"]
+        if v is not None and v not in valid_config:
+            raise DagsterInvalidConfigError(
+                "Snowflake Resource: 'paramstyle' configuration value must be one of:"
+                f" {','.join(valid_config)}"
+            )
+        return v
+
+    @validator("connector")
+    def validate_connector(cls, v):
+        if v is not None and v != "sqlalchemy":
+            raise DagsterInvalidConfigError(
+                "Snowflake Resource: 'connector' configuration value must be None or sqlalchemy"
+            )
+        return v
+
+    @overload
+    def __init__(self, schema: str, *args, **kwargs):
+        ...
+
+    @overload
+    def __init__(self, *args, **kwargs):
+        ...
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+
```

Comment:
This is super painful and I don't think we can recommend this. What are the other options here?